### PR TITLE
Add participant summary card

### LINKED ISSUE #1217: 🤸‍♀️ Participants summary card
## Description
A user should be able to see a summary of the participation at an event, at the top of the participants tab, and take common actions from there.

Also see epic #1214 for context.

## Relevant Job Stories
* When I’m using events to organize basic activism like handing out flyers or putting up posters, I want to make sure that at least one of the people signed up for the event has done the work before and assign them as the contact for the rest of the activists, so that I can feel comfortable that they will be able to run the event
* When I’m using events to organize basic activism and one of the actvists who signed up would be suitable for running the event, I want to reach out to them, so that I can make sure they are able to do so
* When certain events have taken place, I want to feedback into the system who actually showed up, so that I can keep track of attendance and search accordingly
* When I’m using events to organize basic activism and I have done all of the preparations (assigning a contact etc), I want to be able to send out a reminder e-mail, so that all of the attendees get the information they need

## Prerequisites
None

## Requirements
- [ ] The top left should have the title "Participants"
- [ ] The top right should have the current participant count (not including sign-ups) and minimum required participant count
- [ ] The top right should also have a cogwheel to open a popper where the minimum required participant count can be changed
- [ ] There should be a column for "Pending sign-ups"
  - [ ] The column displays the number of people who have signed up but have not been added as participants
  - [ ] When there are >0 pending sign-ups, display a button that when clicked converts all sign-ups to participants
- [ ] There should be a column for "Notifications"
  - [ ] The column displays the number of booked participants that have received notifications (R), against the total number of participants (P)
  - [ ] When R < P, and there is a contact person set, display a button that when clicked sends reminders to the ones who have not yet been reminded

The third column, "Cancelled" is tracked elsewhere

## Possible implementations
The popper for setting the minimum required participant count is already implemented as part of the participants card on the overview tab, added in PR #1196.

"Convert" sign-ups to participants by adding participants the same way as in #1216.

Send all (unsent) reminders by posting to `/orgs/{org_id}/actions/{action_id}/reminders`.

### In situ on page
image

The card in question is the one marked with a blue frame in the screenshot below.
image

### First sign-ups (no participants, 7 sign-ups)
image

### Reminders disabled (no contact person)
image

### Reminders enabled (contact person defined)
image

### More sign-ups (note total counter)
image

### Fully booked
image

## Open questions
None

### PR DESCRIPTION
**Pull request by theo8theo8**
## Description
This PR adds a summary card for event participants in the participant tab.

## Screenshots

Notifications can be sent if a contact person is available
image

If no contact person is available
image

Notifications sent
image

Pending sign-ups bookable
image

## Changes
* Adds functionality for sending reminders to the API
* Adds functionality for loading all respondents on a specific event
* Adds functionality for converting a respondent to a participant
* Adds an overview card in accordance with #1217 
* Changes participant tab to include the new card

## Notes to reviewer
I chose to add the localised messages for this specific tab, even though a lot of them are duplicates of the ones in `eventParticipantsCard`. Not sure if this is the correct way to do it.

## Related issues
Resolves #1217

### CODE REVIEW

**Comment by theo8theo8** [comment]
Not sure what I managed to do that made that test not work, it doesn't look like anything I have touched.

**Comment by ziggabyte** [comment]
> Not sure what I managed to do that made that test not work, it doesn't look like anything I have touched.

Sometimes the tests are just grumpy in CI, if a test that seems completely unrelated fails, I just re-run and hopefully it passes that time. So I re-ran them now and they passed! :D

File: src/features/events/components/ParticipantSummaryCard.tsx
Comment (ziggabyte) [inline src/features/events/components/ParticipantSummaryCard.tsx:177]: I didn't know you could access the palette colours like this in sx, that's super convenient 😄

File: src/features/events/components/ParticipantSummaryCard.tsx
Comment (ziggabyte) [inline src/features/events/components/ParticipantSummaryCard.tsx:35]: I'm thinking there could be use for setting up a route on the "frontend API" to get these numbers (req, avail, reminded, signed) directly, because I think they will be needed in more places than this and we could put these calculations in one place and be re-used in many places? But I'm going to leave that to @richardolsson to weigh in on :D

**Review by ziggabyte — APPROVED** [review]
WOop WOop anothe super nice PR from you! Some small detail comments, but approving because as far as I can see it does what it should :)

File: src/features/events/components/ParticipantSummaryCard.tsx
Comment (richardolsson) [reply src/features/events/components/ParticipantSummaryCard.tsx:177]: This color should not be blue. I know it is blue in the design mockups, but that's because design mockups use another theme (very confusing, I know). The color should just be the default, so these can be taken out.

File: src/features/events/components/ParticipantSummaryCard.tsx
Comment (richardolsson) [reply src/features/events/components/ParticipantSummaryCard.tsx:35]: I think this is a good idea, but requires some more thinking, so lets revisit that separately and not block this PR for it.

The logic looks correct though, so as a first step I would just move it to a model where it can be reused, and I know @rebecarubio is working on something related and could probably take care of that after this is merged.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Looks great! Only think I'd change is the button color which should be the default (primary) color, even though it appears blue in the mockups (very confusing, I know).

**Comment by theo8theo8** [comment]
@richardolsson Could I get a re-run on that failed CI :)?

**Review by richardolsson — APPROVED** [review]
Nice work! Let's merge this. 🎉

**Pull request merged**
